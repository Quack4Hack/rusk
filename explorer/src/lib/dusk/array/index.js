export { default as arraySum } from "./arraySum";
export { default as arraySumByKey } from "./arraySumByKey";
