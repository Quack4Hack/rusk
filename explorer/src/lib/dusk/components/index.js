export { default as Anchor } from "./anchor/Anchor.svelte";
export { default as Select } from "./select/Select.svelte";
export { default as NavList } from "./nav-list/NavList.svelte";
