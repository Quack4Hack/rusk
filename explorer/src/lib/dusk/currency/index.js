export { default as createCompactFormatter } from "./createCompactFormatter";
export { default as createCurrencyFormatter } from "./createCurrencyFormatter";
export { default as currencies } from "./currencies";
