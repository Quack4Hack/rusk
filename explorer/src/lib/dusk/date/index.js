export { default as unixTsToDate } from "./unixTsToDate";
