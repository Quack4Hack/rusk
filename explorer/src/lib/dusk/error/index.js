export { default as getErrorFrom } from "./getErrorFrom";
