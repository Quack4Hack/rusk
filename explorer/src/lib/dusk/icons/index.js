export { default as duskIcon } from "./logo.js";
