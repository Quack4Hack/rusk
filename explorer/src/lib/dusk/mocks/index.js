export { default as IntersectionObserver } from "./IntersectionObserver";
export { default as ResizeObserver } from "./ResizeObserver";
