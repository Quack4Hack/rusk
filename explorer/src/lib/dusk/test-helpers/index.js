export { default as renderWithSimpleContent } from "./renderWithSimpleContent";
export { default as renderWithSlots } from "./renderWithSlots";
export { default as SlotContent } from "./SlotContent.svelte";
