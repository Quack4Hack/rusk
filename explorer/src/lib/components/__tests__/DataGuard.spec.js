import { afterEach, describe, expect, it } from "vitest";
import { cleanup, render } from "@testing-library/svelte";

import { renderWithSimpleContent } from "$lib/dusk/test-helpers";

import { DataGuard } from "..";

describe("DataGuard", () => {
  const baseProps = {
    data: null,
  };
  const baseOptions = {
    props: baseProps,
    target: document.body,
  };

  afterEach(cleanup);

  it("should render the `DataGuard` with the placeholder if no data is passed", () => {
    const { container } = render(DataGuard, baseOptions);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render the `DataGuard` if data is present", () => {
    const data = 1;
    const renderWithSlots = renderWithSimpleContent(DataGuard, {
      ...baseOptions,
      props: { ...baseProps, data },
    });

    expect(renderWithSlots.container.firstChild).toMatchSnapshot();
  });
});
