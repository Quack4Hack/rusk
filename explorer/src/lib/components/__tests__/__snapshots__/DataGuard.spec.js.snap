// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataGuard > should render the \`DataGuard\` if data is present 1`] = `
<span>
  some text
</span>
`;

exports[`DataGuard > should render the \`DataGuard\` with the placeholder if no data is passed 1`] = `
<span>
  - - -
</span>
`;
